src/ui/index.js

```diff
@@ -8,6 +8,13 @@ import UiExports from './ui_exports';
 import UiBundle from './ui_bundle';
 import UiBundleCollection from './ui_bundle_collection';
 import UiBundlerEnv from './ui_bundler_env';
+import i18nPlugin from '../core_plugins/i18n/server/i18n/index';
+import langParser from 'accept-language-parser';
+
+let kibanaTranslations = [];
+let acceptLanguages = '';
```

Comment:
There's potential for a race condition here since every request is going to share this variable. Every request needs its own local copy of `acceptLanguages`, which means we probably need to add an extra param to renderApp. Rather than passing `acceptLanguages` itself, it's probably better to determine the correct locale in the route method, and then pass the locale to `renderApp`.